src/pbt/__init__.py

```diff
@@ -30,104 +43,233 @@ def __init__(self, path_root: str):
+                elif self.project_language == 'scala':
+                    unit_test_results[path_pipeline] = self.test_scala(path_pipeline_absolute)
+            is_any_ut_failed = False
+            for path_pipeline, return_code in unit_test_results.items():
+                if return_code not in (0, 5):
+                    is_any_ut_failed = True
+                    print(
+                        f"\n[bold red]❌ Unit test for pipeline: {path_pipeline} failed with return code {return_code}![/bold red]")
+                else:
+                    print(f"\n[bold blue]✅ Unit test for pipeline: {path_pipeline} succeeded.[/bold blue]")
+
+            if is_any_ut_failed:
+                sys.exit(1)
+        else:
+            sys.exit(1)
+
+    def build_python(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [
+                # Install dependencies of particular pipeline
+                # 1. Export dependencies to egg_info/requires.txt
+                Process(['python3', 'setup.py', '-q', 'egg_info'], path_pipeline_absolute),
+                # 2. Extract the install_requires, tests_requires and extra_requires and install them
+                Process([f"pip3 install -q `grep -v '^\[' *.egg-info/requires.txt`"],
+                        path_pipeline_absolute, is_shell=True),
+                # Check for compilation errors
+                Process(['python3', '-m', 'compileall', '.', '-q'], path_pipeline_absolute),
+                # Generate wheel
+                Process(['python3', 'setup.py', 'bdist_wheel'], path_pipeline_absolute)
+            ]
+        )
+
+    def build_scala(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [Process(['mvn', 'clean', 'package', '-q', '-DskipTests'], path_pipeline_absolute)])
+
+    def test_scala(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [
+                Process(['mvn', 'test', '-q', '-Dfabric=' + self.fabric.strip()],
+                        path_pipeline_absolute)
+            ]
+        )
+
+    def test_python(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [
+                # Install dependencies of particular pipeline
+                # 1. Export dependencies to egg_info/requires.txt
+                Process(['python3', 'setup.py', '-q', 'egg_info'], path_pipeline_absolute),
+                # 2. Extract the install_requires, tests_requires and extra_requires and install them
+                Process([f"pip3 install -q `grep -v '^\[' *.egg-info/requires.txt`"],
+                        path_pipeline_absolute, is_shell=True),
+                # Run the unit test
+                Process(['python3', '-m', 'pytest', '-v', 'test/TestSuite.py', '--html=report.html',
+                         '--self-contained-html', '--junitxml=report.xml'],
+                        path_pipeline_absolute)
```

Comment:
We don't need `html and junit` reporting (if console is going to show all logs).
Plus you'll have to install additional packages of `pytest` to run tests using pytest command

Reply:
Removed html and junit. I had initially installed things like pytest-html in the setup github actions file.